lib/util/generic_type.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:model_generator/util/type_checker.dart';
+
+class DartType {
+  final String baseType;
+  final List<DartType> generics;
+
+  factory DartType(String type) {
+    if (!type.contains('<')) return DartType._(type.trim(), []);
+
+    final start = type.indexOf('<');
+    final end = type.lastIndexOf('>');
+    final typeArgumentList = type.substring(start + 1, end);
+    final name = type.substring(0, start).trim();
+    final arguments = _splitByComma(typeArgumentList);
+
+    return DartType._(
+        name, arguments.map((e) => DartType(e)).toList(growable: false));
+  }
+
+  DartType._(this.baseType, this.generics);
+
+  Iterable<String> get leaves {
+    if (generics.isEmpty) return [baseType];
+    final leaves = <String>{};
+    generics.forEach((element) => leaves.addAll(element.leaves));
+    return leaves;
+  }
+
+  void checkTypesKnown(final Set<String> names) {
+    if (generics.isEmpty) {
+      if (!TypeChecker.isKnownDartType(baseType) && !names.contains(baseType)) {
+        throw Exception(
+            'Could not generate all models. `$baseType` is not added to the config file');
+      }
+    } else if (!TypeChecker.isKnownDartCollectionType(baseType)) {
+      throw Exception(
+          'Could not generate all models. `$baseType` is currently not supported as generic base type');
+    } else {
+      generics.forEach((e) => e.checkTypesKnown(names));
+    }
+  }
+
+  @override
+  String toString() {
+    if (generics.isEmpty) {
+      return baseType;
+    }
+    final builder = StringBuffer(baseType)..write('<');
+    var c = 0;
+    generics.forEach((element) {
+      if (c++ > 0) builder.write(', ');
+      builder.write(element.toString());
+    });
+    builder.write('>');
+    return builder.toString();
+  }
+}
+
+List<String> _splitByComma(String typeArgumentList) {
+  final strings = <String>[];
+  final stringBuffer = StringBuffer();
+  var level = 0;
+
+  for (var i = 0; i < typeArgumentList.length; ++i) {
+    final c = typeArgumentList.codeUnitAt(i);
+    if (c == 0x2C && level == 0) {
+      strings.add(stringBuffer.toString().trim());
+      stringBuffer.clear();
+    } else {
+      stringBuffer.write(String.fromCharCode(c));
+    }
+
+    if (c == 0x3C) {
```

Comment:
I would move this to a const variable. So it is clear to read what is happening here.